publishers/community/slack/slack_layout.py

```diff
@@ -0,0 +1,268 @@
+
+        author = rule_presentation['author']
+
+        return {
+            'slack.text': 'Rule triggered',
+            'slack.attachments': [
+                {
+                    'fallback': 'Rule triggered: {}'.format(rule_name),
+                    'color': self._color(),
+                    'author_name': author,
+                    'author_link': self._author_url(author),
+                    'author_icon': self._author_icon(author),
+                    'title': rule_name,
+                    'title_link': self._title_url(rule_name),
+                    'text': cgi.escape(rule_presentation['description']),
+                    'image_url': '',
+                    'thumb_url': '',
+                    'footer': '',
+                    'footer_icon': '',
+                    'ts': time.mktime(alert.created.timetuple()) if alert.created else '',
+                    'mrkdwn_in': [],
+                },
+            ],
+
+            # This information is passed-through to future publishers.
+            '_previous_publication': publication,
+        }
+
+    @staticmethod
+    def _color():
+        """The color of this section"""
+        return RAUSCH
+
+    @classmethod
+    def _author_url(cls, _):
+        """When given an author name, returns a clickable link, if any"""
+        return ''
+
+    @classmethod
+    def _author_icon(cls, _):
+        """When given an author name, returns a URL to an icon, if any"""
+        return ''
+
+    @classmethod
+    def _title_url(cls, rule_name):
+        """When given the rule_name, returns a clickable link, if any"""
+
+        # It's actually super hard to generate a exact link to a file just from the rule_name,
+        # because the rule/ directory files are not deployed with the publishers in the alert
+        # processor.
+        # Instead, we send them to Github with a formatted query string that is LIKELY to
+        # find the correct file.
+        #
+        # If you do not want URLs to show up, simply override this method and return empty string.
+        return '{}{}?{}'.format(
+            cls._GITHUB_REPO_URL,
+            cls._SEARCH_PATH,
+            urllib.urlencode({
+                'q': '{} path:{}'.format(rule_name, cls._RULES_PATH)
+            })
+        )
+
+
+@Register
+class AttachRuleInfo(AlertPublisher):
+    """This publisher adds a slack attachment with fields from the rule's description
+
+    It can include such fields as "reference" or "playbook" but will NOT include the description
+    or the author.
+    """
+
+    def publish(self, alert, publication):
+        publication['slack.attachments'] = publication.get('slack.attachments', [])
+
+        rule_description = alert.rule_description
+        rule_presentation = RuleDescriptionParser.present(rule_description)
+
+        publication['slack.attachments'].append({
+            'color': self._color(),
+            'fields': map(
```

Comment:
this might just be me, but I think the map usage can lead to confusion

Reply:
Changed to 

```
'fields': [
  {'title': key.capitalize(), 'value': rule_presentation['fields'][key]}
  for key in rule_presentation['fields'].keys()
],
```